integrations/optimum/src/haystack_integrations/components/embedders/optimum_text_embedder.py

```diff
@@ -0,0 +1,200 @@
+        token = token.resolve_value() if token else None
+
+        self.prefix = prefix
+        self.suffix = suffix
+        self.normalize_embeddings = normalize_embeddings
+        self.onnx_execution_provider = onnx_execution_provider
+
+        model_kwargs = model_kwargs or {}
+
+        # Check if the model_kwargs contain the parameters, otherwise, populate them with values from init parameters
+        model_kwargs.setdefault("model_id", model)
+        model_kwargs.setdefault("provider", onnx_execution_provider)
+        model_kwargs.setdefault("use_auth_token", token)
+
+        self.model_kwargs = model_kwargs
+        self.embedding_model = None
+        self.tokenizer = None
+
+    def warm_up(self):
+        """
+        Convert the model to ONNX.
+
+        The model is cached if the "TensorrtExecutionProvider" is used, since it takes a while to build the TensorRT
+        engine.
+        """
+        if self.embedding_model is None:
+            self.tokenizer = AutoTokenizer.from_pretrained(self.model)
+
+            if self.onnx_execution_provider == "TensorrtExecutionProvider":
+                # Cache engine for TensorRT
+                provider_options = {
+                    "trt_engine_cache_enable": True,
+                    "trt_engine_cache_path": f"tmp/trt_cache_{self.model}",
+                }
+                self.embedding_model = ORTModelForFeatureExtraction.from_pretrained(
+                    **self.model_kwargs, use_cache=False, provider_options=provider_options
+                )
+            else:
+                # export=True converts the model to ONNX on the fly
+                self.embedding_model = ORTModelForFeatureExtraction.from_pretrained(**self.model_kwargs, export=True)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Serialize this component to a dictionary.
+        """
+        serialization_dict = default_to_dict(
+            self,
+            model=self.model,
+            prefix=self.prefix,
+            suffix=self.suffix,
+            normalize_embeddings=self.normalize_embeddings,
+            onnx_execution_provider=self.onnx_execution_provider,
+            model_kwargs=self.model_kwargs,
+            token=self.token.to_dict() if self.token else None,
+        )
+
+        model_kwargs = serialization_dict["init_parameters"]["model_kwargs"]
+        model_kwargs.pop("token", None)
+
+        serialize_hf_model_kwargs(model_kwargs)
+        return serialization_dict
+
+    @classmethod
+    def from_dict(cls, data: Dict[str, Any]) -> "OptimumTextEmbedder":
+        """
+        Deserialize this component from a dictionary.
+        """
+        deserialize_secrets_inplace(data["init_parameters"], keys=["token"])
+        deserialize_hf_model_kwargs(data["init_parameters"]["model_kwargs"])
+        return default_from_dict(cls, data)
+
+    def mean_pooling(self, model_output: torch.tensor, attention_mask: torch.tensor) -> torch.tensor:
```

Comment:
Another potential issue I see. The default model BAAI/bge-small-en-v1.5 shows in their code sample on HF that we should be using cls pooling and not mean pooling. I think we need to allow the user to specify which pooling method they would like to use since it can often change with the picked model. 

See here: https://huggingface.co/BAAI/bge-small-en-v1.5#using-huggingface-transformers

Relevant code snippet:
```python
from transformers import AutoTokenizer, AutoModel
import torch
# Sentences we want sentence embeddings for
sentences = ["样例数据-1", "样例数据-2"]

# Load model from HuggingFace Hub
tokenizer = AutoTokenizer.from_pretrained('BAAI/bge-large-zh-v1.5')
model = AutoModel.from_pretrained('BAAI/bge-large-zh-v1.5')
model.eval()

# Tokenize sentences
encoded_input = tokenizer(sentences, padding=True, truncation=True, return_tensors='pt')
# for s2p(short query to long passage) retrieval task, add an instruction to query (not add instruction for passages)
# encoded_input = tokenizer([instruction + q for q in queries], padding=True, truncation=True, return_tensors='pt')

# Compute token embeddings
with torch.no_grad():
    model_output = model(**encoded_input)
    # Perform pooling. In this case, cls pooling.
    sentence_embeddings = model_output[0][:, 0]
# normalize embeddings
sentence_embeddings = torch.nn.functional.normalize(sentence_embeddings, p=2, dim=1)
print("Sentence embeddings:", sentence_embeddings)
```

Reply:
If the model is on the HF Hub has the necessary Sentence Transformer files (e.g. https://huggingface.co/BAAI/bge-small-en-v1.5/blob/main/1_Pooling/config.json) then it would be possible to auto detect with pooling method to use.

Reply:
Yes, we can do that. I have added a `pooling` submodule with classes to fetch the sentence-transformers config from HF Hub. The other pooling methods (CLS Pooling, Max Pooling, Mean Square Root - Length Pooling, Weighted Mean Pooling and Last Token Pooling) are also now supported. 

The implementation for the pooling methods has been adapted from the [sentence-transformers pooling implementation](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/models/Pooling.py).